Rakefile.rb

```diff
@@ -48,6 +48,36 @@ def name_from(json)
   orgs.first[:name]
 end
 
+def file_to_commit_metadata
+  # There's about 2MB of data returned by this command, so rather than
+  # using backticks to turning it into a string and then splitting it,
+  # parse the output line-by-line. As a bonus this doesn't
+  # unnecessarily invoke a shell.
+  command = ['git', '--no-pager', 'log', '--name-only',  '--format=%h|%at', '--', 'data']
+  result = {}
+  sha, timestamp = nil
+  Kernel.open('|-', 'r') { |f|
```

Comment:
I think you're right, there's no advantage to the more explicitly fork-and-exec code here, so I've switched it to use `IO.popen`; indeed it's much easier to read. Fixed in: 7afd601f861d8b4634